internal/controllers/reconciliation/patch_test.go

```diff
@@ -299,3 +300,164 @@ func TestPatchDeletionBeforeUpgrade(t *testing.T) {
 	require.NotEqual(t, creationUID, recreationUID)
 	require.Equal(t, val, cm.Data[key])
 }
+
+// TestPatchDeletionForResourceWithReconciliation proves that a patch resource can delete the resource it references
+// even the resource with reconcile-interval annotation and it won't be re-created again.
+func TestPatchDeletionForResourceWithReconciliation(t *testing.T) {
+	scheme := runtime.NewScheme()
+	corev1.SchemeBuilder.AddToScheme(scheme)
+	testv1.SchemeBuilder.AddToScheme(scheme)
+
+	ctx := testutil.NewContext(t)
+	mgr := testutil.NewManager(t)
+	upstream := mgr.GetClient()
+	downstream := mgr.DownstreamClient
+
+	// Test configmap setup
+	cmName := "test-obj"
+	cmNamespace := "default"
+	cm := &corev1.ConfigMap{}
+	cm.Name = cmName
+	cm.Namespace = cmNamespace
+	cm.Annotations = map[string]string{
+		"eno.azure.io/readiness-group":    "2",
+		"eno.azure.io/reconcile-interval": "1ms",
+	}
+
+	registerControllers(t, mgr)
+	testutil.WithFakeExecutor(t, mgr, func(ctx context.Context, s *apiv1.Synthesizer, input *krmv1.ResourceList) (*krmv1.ResourceList, error) {
+		obj := &unstructured.Unstructured{
+			Object: map[string]any{
+				"apiVersion": "eno.azure.io/v1",
+				"kind":       "Patch",
+				"metadata": map[string]any{
+					"name":      cmName,
+					"namespace": cmNamespace,
+					"annotations": map[string]string{
+						// This patch should be applied before the configmap is re-created.
```

Comment:
reconcile-interval annotation won't work for patch if i understand it correctly, it won't re-queue due to the resource.Deleted() always be true for deletion patch https://github.com/Azure/eno/blob/c2af2ca0b63171213a0f98aa2a7c3b91f70000f0/internal/resource/resource.go#L63

Reply:
Wow this is not what I expected at all 😆 . I actually think this behavior makes sense, but we should probably document it somewhere, at some point. Since it does kind of violate the semantics of the reconcile-interval annotation.